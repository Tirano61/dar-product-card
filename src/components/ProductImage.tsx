import React from "react";
import { useContext } from "react";
import { ProductContext } from "./ProductCard";

import styles from "../styles/styles.module.css";
import noImage from "../assets/no-image.jpg";

interface Props {
  img?: string,
  className?: string
  style?: React.CSSProperties
}

export const ProductImage = ({ img = '', className, style }: Props) => {
  const { product } = useContext(ProductContext);
  let imgShow: string;
  if (img) {
    imgShow = img;
  } else if (product.img) {
    imgShow = product.img;
  } else {
    imgShow = noImage;
  }
  return (
    <img
      className={`${styles.productImg} ${className}`}
      style={style}
      src={imgShow}
      alt="Product image"
    />
  )
}