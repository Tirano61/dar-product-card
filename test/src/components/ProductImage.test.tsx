import React from "react";
import { render } from "@testing-library/react";
import { ProductCard, ProductImage } from '../../../src/components';
import { product2 } from "../data/products";



describe('ProductImage', () => {
  test('Debe mostrar el componente correctamente con la imagen personaliado', () => {
    const wrapper = render(
      <ProductImage img="url de la imagen" />
    )

    expect(wrapper.asFragment()).toMatchSnapshot();

  });
  test('Debe mostrar el conponente con la imagenu del producto', () => {
    const wrapper = render(
      <ProductCard product={product2} >
        {
          () => (
            <ProductImage />
          )
        }
      </ProductCard>
    )

    expect(wrapper.asFragment()).toMatchSnapshot();
  });

});