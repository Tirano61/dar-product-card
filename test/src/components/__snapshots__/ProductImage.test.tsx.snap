// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe mostrar el componente correctamente con la imagen personaliado 1`] = `
<DocumentFragment>
  <img
    alt="Product image"
    class="productImg undefined"
    src="url de la imagen"
  />
</DocumentFragment>
`;

exports[`ProductImage Debe mostrar el conponente con la imagenu del producto 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <img
      alt="Product image"
      class="productImg undefined"
      src="./coffee-mug.png"
    />
  </div>
</DocumentFragment>
`;
